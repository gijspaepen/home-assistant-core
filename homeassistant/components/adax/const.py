"""Constants for the Adax integration."""

from typing import Final

from homeassistant.const import Platform

ACCOUNT_ID: Final = "account_id"
CLOUD = "Cloud"
CONNECTION_TYPE = "connection_type"
DOMAIN: Final = "adax"
LOCAL = "Local"
WIFI_SSID = "wifi_ssid"
WIFI_PSWD = "wifi_pswd"

PLATFORMS = [Platform.CLIMATE, Platform.SENSOR]
