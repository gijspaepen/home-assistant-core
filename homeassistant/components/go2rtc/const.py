"""Go2rtc constants."""

DOMAIN = "go2rtc"

CONF_DEBUG_UI = "debug_ui"
EXCLUSE_ERROR_MESSAGE = "Url and debug_ui cannot be set at the same time."
