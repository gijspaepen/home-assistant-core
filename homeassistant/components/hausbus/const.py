"""Constants for the Haus-Bus integration."""

from __future__ import annotations

import logging

LOGGER = logging.getLogger(__package__)

DOMAIN = "hausbus"
ATTR_ON_STATE = "on_state"
