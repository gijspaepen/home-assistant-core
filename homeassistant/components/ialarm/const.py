"""Constants for the iAlarm integration."""

from pyialarm import IAlarm

from homeassistant.components.alarm_control_panel import AlarmControlPanelEntityState

DATA_COORDINATOR = "ialarm"

DEFAULT_PORT = 18034

DOMAIN = "ialarm"

IALARM_TO_HASS = {
    IAlarm.ARMED_AWAY: AlarmControlPanelEntityState.ARMED_AWAY,
    IAlarm.ARMED_STAY: AlarmControlPanelEntityState.ARMED_HOME,
    IAlarm.DISARMED: AlarmControlPanelEntityState.DISARMED,
    IAlarm.TRIGGERED: AlarmControlPanelEntityState.TRIGGERED,
}
