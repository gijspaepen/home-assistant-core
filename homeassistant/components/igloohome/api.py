class Api:  # noqa: D100, D101
    def __init__(self) -> None:  # noqa: D107
        pass
