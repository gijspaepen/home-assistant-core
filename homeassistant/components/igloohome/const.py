"""Constants for the igloohome integration."""

DOMAIN = "igloohome"

OAUTH2_AUTHORIZE = "https://auth.igloohome.co/login"
OAUTH2_TOKEN = "https://auth.igloohome.co/oauth2/token"
