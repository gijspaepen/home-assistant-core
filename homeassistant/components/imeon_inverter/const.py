"""Constant for Imeon component."""

DOMAIN = "imeon_inverter"
HUBNAME = "imeon_inverter_hub"
TIMEOUT = 10  # seconds
