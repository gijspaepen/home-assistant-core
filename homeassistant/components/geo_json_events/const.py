"""Define constants for the GeoJSON events integration."""

from __future__ import annotations

from typing import Final

from homeassistant.const import Platform

DOMAIN: Final = "geo_json_events"

PLATFORMS: Final = [Platform.GEO_LOCATION]

ATTR_EXTERNAL_ID: Final = "external_id"
DEFAULT_RADIUS_IN_KM: Final = 20.0
DEFAULT_RADIUS_IN_M: Final = 20000.0
DEFAULT_UPDATE_INTERVAL: Final = 300
SOURCE: Final = "geo_json_events"
CONF_UPDATE_INTERVAL: Final = "update_interval"

SIGNAL_DELETE_ENTITY: Final = "geo_json_events_delete_{}"
SIGNAL_UPDATE_ENTITY: Final = "geo_json_events_update_{}"
