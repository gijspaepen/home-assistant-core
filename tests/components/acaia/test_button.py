"""Tests for the acaia buttons."""

from datetime import timedelta
from unittest.mock import MagicMock

from freezegun.api import FrozenDateTimeFactory
import pytest
from syrupy import SnapshotAssertion

from homeassistant.components.button import DOMAIN as BUTTON_DOMAIN, SERVICE_PRESS
from homeassistant.const import ATTR_ENTITY_ID, STATE_UNAVAILABLE, STATE_UNKNOWN
from homeassistant.core import HomeAssistant
from homeassistant.helpers import entity_registry as er

from tests.common import async_fire_time_changed

pytestmark = pytest.mark.usefixtures("init_integration")


BUTTONS = (
    "tare",
    "reset_timer",
    "start_stop_timer",
)


async def test_buttons(
    hass: HomeAssistant,
    entity_registry: er.EntityRegistry,
    snapshot: SnapshotAssertion,
) -> None:
    """Test the acaia buttons."""
    for button in BUTTONS:
        state = hass.states.get(f"button.lunar_ddeeff_{button}")
        assert state
        assert state == snapshot(name=f"state_button_{button}")

        entry = entity_registry.async_get(state.entity_id)
        assert entry
        assert entry == snapshot(name=f"entry_button_{button}")


async def test_button_presses(
    hass: HomeAssistant,
    mock_scale: MagicMock,
) -> None:
    """Test the acaia button presses."""

    for button in BUTTONS:
        await hass.services.async_call(
            BUTTON_DOMAIN,
            SERVICE_PRESS,
            {
                ATTR_ENTITY_ID: f"button.lunar_ddeeff_{button}",
            },
            blocking=True,
        )

        function = getattr(mock_scale, button)
        function.assert_called_once()


async def test_buttons_unavailable_on_disconnected_scale(
    hass: HomeAssistant,
    mock_scale: MagicMock,
    freezer: FrozenDateTimeFactory,
) -> None:
    """Test the acaia buttons are unavailable when the scale is disconnected."""

    for button in BUTTONS:
        state = hass.states.get(f"button.lunar_ddeeff_{button}")
        assert state
        assert state.state == STATE_UNKNOWN

    mock_scale.connected = False
    freezer.tick(timedelta(minutes=10))
    async_fire_time_changed(hass)
    await hass.async_block_till_done()

    for button in BUTTONS:
        state = hass.states.get(f"button.lunar_ddeeff_{button}")
        assert state
        assert state.state == STATE_UNAVAILABLE
