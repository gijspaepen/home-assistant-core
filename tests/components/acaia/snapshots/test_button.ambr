# serializer version: 1
# name: test_button_presses[entry_button_reset_timer]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'button',
    'entity_category': None,
    'entity_id': 'button.lunar_123456_reset_timer',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Reset timer',
    'platform': 'acaia',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'reset_timer',
    'unique_id': 'aa:bb:cc:dd:ee:ff_reset_timer',
    'unit_of_measurement': None,
  })
# ---
# name: test_button_presses[entry_button_start_stop_timer]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'button',
    'entity_category': None,
    'entity_id': 'button.lunar_123456_start_stop_timer',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Start/stop timer',
    'platform': 'acaia',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'start_stop',
    'unique_id': 'aa:bb:cc:dd:ee:ff_start_stop',
    'unit_of_measurement': None,
  })
# ---
# name: test_button_presses[entry_button_tare]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'button',
    'entity_category': None,
    'entity_id': 'button.lunar_123456_tare',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Tare',
    'platform': 'acaia',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'tare',
    'unique_id': 'aa:bb:cc:dd:ee:ff_tare',
    'unit_of_measurement': None,
  })
# ---
# name: test_button_presses[state_button_reset_timer]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'LUNAR_123456 Reset timer',
    }),
    'context': <ANY>,
    'entity_id': 'button.lunar_123456_reset_timer',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_button_presses[state_button_start_stop_timer]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'LUNAR_123456 Start/stop timer',
    }),
    'context': <ANY>,
    'entity_id': 'button.lunar_123456_start_stop_timer',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_button_presses[state_button_tare]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'LUNAR_123456 Tare',
    }),
    'context': <ANY>,
    'entity_id': 'button.lunar_123456_tare',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
